document-store/src/main/java/org/hypertrace/core/documentstore/Collection.java

```diff
@@ -282,18 +285,21 @@ Optional<Document> update(
    * documents would match.
    *
    * @param query The query to be executed. Also, contains the filter for updating
-   * @param updates The list of sub-document updates to be performed
+   * @param updates The collection of sub-document updates to be performed in any order
    * @param updateOptions Options for updating/returning the document
    * @return A closeable iterator to the old (before update) or new (after update) documents if
    *     requested, otherwise an empty iterator.
-   * @throws IOException if there was any error in updating/fetching the documents or the updates is
-   *     empty
+   * @throws IOException if there was any error in updating/fetching the documents or argument
+   *     validation failed (like no updates passed)
    * @implSpec The definition of an update here is
    *     <ol>
    *       <li>The existing sub-documents will be updated
    *       <li>New sub-documents will be created if they do not exist
    *       <li>None of the existing sub-documents will be removed
    *     </ol>
+   *     <p>If more than one update is modifying the same hierarchy (e.g.: a->b and a->b->c), should
```

Comment:
I think, it's up to the client, isn't it? We can discuss this offline, I am not fully clear here. For now, this is not a blocker.